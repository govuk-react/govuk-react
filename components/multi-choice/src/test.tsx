import * as React from 'react';
import { mount } from 'enzyme';

import { MultiChoice } from '.';

describe('MultiChoice', () => {
  it('renders with a label', () => {
    const wrapper = mount(<MultiChoice label="Example label">Example</MultiChoice>);
    expect(wrapper.find('LabelText').contains('Example label')).toBe(true);
  });

  it('renders with a hint', () => {
    const wrapper = mount(
      <MultiChoice label="Example label" hint="Example hint">
        Example
      </MultiChoice>
    );
    expect(wrapper.find('HintText').contains('Example hint')).toBe(true);
  });

  it('renders with an error', () => {
    const wrapper = mount(
      <MultiChoice label="Example label" meta={{ touched: true, error: 'Example error' }}>
        Example
      </MultiChoice>
    );
    expect(wrapper.find('ErrorText').contains('Example error')).toBe(true);
  });

  it('matches snapshot', () => {
    const metaData = { touched: true, error: 'Example error' };
    const wrapper = mount(
      <MultiChoice label="Example label" hint="Example hint" meta={metaData}>
        Label
      </MultiChoice>
    );
    expect(wrapper).toMatchSnapshot();
  });
});
