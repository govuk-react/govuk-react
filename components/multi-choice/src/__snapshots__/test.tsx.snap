// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MultiChoice matches snapshot 1`] = `
.c1 {
  font-family: "nta",Arial,sans-serif;
  -webkit-font-smoothing: antialiased;
  -moz-osx-font-smoothing: grayscale;
  font-weight: 400;
  font-size: 16px;
  line-height: 1.25;
  color: #0b0c0c;
  display: block;
  clear: none;
  padding-bottom: 2px;
  margin-bottom: 0;
}

.c3 {
  font-family: "nta",Arial,sans-serif;
  -webkit-font-smoothing: antialiased;
  -moz-osx-font-smoothing: grayscale;
  font-weight: 700;
  font-size: 16px;
  line-height: 1.25;
  display: block;
  margin-bottom: 15px;
  clear: both;
  color: #d4351c;
}

.c2 {
  font-family: "nta",Arial,sans-serif;
  -webkit-font-smoothing: antialiased;
  -moz-osx-font-smoothing: grayscale;
  font-weight: 400;
  font-size: 16px;
  line-height: 1.25;
  display: block;
  margin-bottom: 15px;
  color: #6f777b;
}

.c0 {
  padding: 0;
  margin: 0;
  border: 0;
  box-sizing: border-box;
  width: 100%;
  border-left: 4px solid #d4351c;
  margin-right: 15px;
  padding-left: 10px;
  margin-bottom: 0;
}

.c0:after {
  content: '';
  display: table;
  clear: both;
}

@media print {
  .c1 {
    font-size: 14px;
    line-height: 1.15;
  }
}

@media only screen and (min-width:641px) {
  .c1 {
    font-size: 19px;
    line-height: 1.3157894736842106;
  }
}

@media print {
  .c1 {
    color: #000;
  }
}

@media only screen and (min-width:641px) {
  .c1 {
    margin-bottom: 0;
  }
}

@media print {
  .c3 {
    font-size: 14px;
    line-height: 1.15;
  }
}

@media only screen and (min-width:641px) {
  .c3 {
    font-size: 19px;
    line-height: 1.3157894736842106;
  }
}

@media print {
  .c2 {
    font-size: 14px;
    line-height: 1.15;
  }
}

@media only screen and (min-width:641px) {
  .c2 {
    font-size: 19px;
    line-height: 1.3157894736842106;
  }
}

@media only screen and (min-width:641px) {
  .c0 {
    margin-bottom: 0;
  }
}

<MultiChoice
  hint="Example hint"
  label="Example label"
  meta={
    Object {
      "error": "Example error",
      "touched": true,
    }
  }
>
  <src__StyledFieldset
    error={true}
  >
    <div
      className="c0"
    >
      <LabelText>
        <span
          className="c1"
        >
          Example label
        </span>
      </LabelText>
      <HintText>
        <span
          className="c2"
        >
          Example hint
        </span>
      </HintText>
      <ErrorText>
        <span
          className="c3"
        >
          Example error
        </span>
      </ErrorText>
      Label
    </div>
  </src__StyledFieldset>
</MultiChoice>
`;
